qt/scientific_interfaces/Direct/ALFView_presenter.cpp

```diff
@@ -0,0 +1,69 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ALFView_presenter.h"
+#include "ALFView_model.h"
+#include "ALFView_view.h"
+
+#include "MantidAPI/FileFinder.h"
+
+#include <functional>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+ALFView_presenter::ALFView_presenter(ALFView_view *view, ALFView_model *model)
+    : m_view(view), m_model(model), m_currentRun(0), m_currentFile(""),
+      m_loadRunObserver(nullptr) {
+  m_loadRunObserver = new VoidObserver();
+  m_model->loadEmptyInstrument();
+}
+
+void ALFView_presenter::initLayout() {
+  // connect to new run
+  m_view->observeLoadRun(m_loadRunObserver);
+  std::function<void()> loadBinder =
+      std::bind(&ALFView_presenter::loadRunNumber, this);
+  m_loadRunObserver->setSlot(loadBinder);
+}
+
+void ALFView_presenter::loadAndAnalysis(const std::string &pathToRun) {
+  try {
+
```

Comment:
Whitespace like this can be jarring when reading. Can we remove it? (Similarly in other cases around).